File: gittools/models/branch.py

from scriptcore.console.basescript import BaseScript
import re


class Branch(object):

    @staticmethod
    def all():
        """
        Get all branches
        :return:    List of Branch's
        """

        current_script = BaseScript.current_script()
        output = current_script.execute('git branch')

        branches = []
        for line in output:
            match = re.match('^(\*?)\s*(.+)$', line)

            name = match.group(2)
            current = match.group(1) == '*'

            if not name:
                continue

            branches.append(Branch(name, current))

        return branches

    def __init__(self, name, current):
        """
        Constructor
        :param name:    Name of the branch
        :param current: Current branch
        """

        self.name = name
        self.current = current

File: src/models/branch.ts
import * as childProcess from 'child_process'

export default class Branch {
  name: string
  current: boolean

  static all (): Branch[] {
    const output = childProcess.execSync('git branch').toString()

    return output.split('\n')
      .map(line => {
        const match = /^(\*?)\s*(.+)$/.exec(line)
        if (match) {
          const name = match[2]
          const current = !!match[1]
          return new Branch(name, current)
        }
      })
      .filter(branch => !!branch) as Branch[]
  }

  constructor (name: string, current: boolean) {
    this.name = name
    this.current = current
  }
}
